calculator.rb

```diff
@@ -0,0 +1,34 @@
+def prompt_operator
+  puts "enter a math operation"
+  operator = gets.chomp
+end
+
+def prompt_number
+  begin
+    puts "enter a number for the math operation"
+    Float(gets.chomp) 
+  rescue ArgumentError
+    puts "invalid number"
+    retry
+  end
+end
+
+x = prompt_number
+y = prompt_number
+operator = prompt_operator
+
+while
```

Comment:
A `while` loop with no condition actually doesn't do anything in Ruby afaik! At the very least, taking out the `while` loop does not change your program in this case.